app/models/task.rb

```diff
@@ -0,0 +1,33 @@
+# == Schema Information
+#
+# Table name: tasks
+#
+#  project_id  :string
+#  name        :string
+#  description :text
+#  state       :integer          default(10)
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#  id          :uuid             not null, primary key
+#
+
+class Task < ActiveRecord::Base
+  belongs_to :project
+  validates :name, presence: true
+  validates :state, presence: true
+  validates :project_id, presence: true
+  after_commit :send_message, on: :update
```

Comment:
Performing actions (especially ones that interact with external services) using ActiveRecord callbacks can lead to problematic situations. It makes things harder to test and also makes the action completely coupled to the idea of persistence. While there are ways to by-pass the callbacks, it's generally good practice to limit the use of them.
An old article I wrote:
http://blog.originate.com/blog/2013/08/27/improving-abstraction-and-testability-by-removing-activerecord-callbacks/

If creating a service seems heavy handy, a decorator can work as well.
```
class NotifiableTask < SimpleDecorator
  def save
    super && send_message
  end 

  def send_message 
    return unless state == 'done'
    TextMessenger.send_text __getobj__
  end
end
```

Then you could do:
`NotifiableTask.new(task).save`